tests/unit/test_main.py

```diff
@@ -1016,3 +1016,670 @@ def test_match_and_apply(self, flow, args, resp):
+                            "action_type": "pop_int"
+                        }, {
+                            "action_type": "set_vlan",
+                            "vlan_id": 201
+                        }, {
+                            "action_type": "output",
+                            "port": 15
+                        }]}
+                    ],
+                    "table_id": 2,
+                    "table_group": "base",
+                    "priority": 20000,
+                    }, },
+                ]}
+        payload = [
+            {"trace": {
+                "switch": {
+                    "dpid": "00:00:00:00:00:00:00:01",
+                    "in_port": 15
+                    },
+                "eth": {"dl_vlan": 201}
+                }}
+        ]
+        mock_interface.link.endpoint_a.port_number = 19
+        mock_interface.link.endpoint_b.port_number = 20
+
+        switch.get_interface_by_port_no.return_value = mock_interface
+
+        resp = await self.api_client.put(self.traces_endpoint, json=payload)
+        assert resp.status_code == 200
+        current_data = resp.json()
+        result = current_data["result"][0]
+
+        assert result[0]['dpid'] == '00:00:00:00:00:00:00:01'
+        assert result[0]['port'] == 15
+        assert result[0]['vlan'] == 201
+
+        assert result[1]['port'] == 20
+        assert result[1]['out']['port'] == 16
+        assert result[1]['out']['vlan'] == 200
+        payload = [
+            {"trace": {
+                "switch": {
+                    "dpid": "00:00:00:00:00:00:00:01",
+                    "in_port": 16
+                    },
+                "eth": {"dl_vlan": 200}
+                }}
+        ]
+
+        mock_interface.link.endpoint_a.port_number = 17
+        mock_interface.link.endpoint_b.port_number = 18
+
+        switch.get_interface_by_port_no.return_value = mock_interface
+
+        resp = await self.api_client.put(self.traces_endpoint, json=payload)
+        assert resp.status_code == 200
+        current_data = resp.json()
+        result = current_data["result"][0]
+
+        assert result[0]['dpid'] == '00:00:00:00:00:00:00:01'
+        assert result[0]['port'] == 16
+        assert result[0]['vlan'] == 200
+
+        assert result[1]['port'] == 18
+        assert result[1]['out']['port'] == 15
+        assert result[1]['out']['vlan'] == 201
+
+    @patch("napps.amlight.sdntrace_cp.main.get_stored_flows")
+    async def test_trace_goto_table_inter(self, mock_stored_flows, event_loop):
+        """Test trace rest call."""
+        self.napp.controller.loop = event_loop
+
+        switch1 = self.napp.controller.switches["00:00:00:00:00:00:00:01"]
+        switch2 = self.napp.controller.switches["00:00:00:00:00:00:00:02"]
+
+        mock_interface12 = Interface("interface A", 1, MagicMock())
+        mock_interface22 = Interface("interface B", 1, MagicMock())
+
+        iface1 = get_interface_mock("", 11, switch1)
```

Comment:
Instead of setting an empty name you can actually set a name such as `s1-eth11` same pattern that we've been using in other unit tests. Empty string won't result in issues here, but also improves the intention and clarity of what's being mocked.

Reply:
I'll also recommend that you include the topology in the docstring since this is a relatively complex unit test, good coverage, just so future readers can easily understand (you'll need to rename sw6 as sw2 which is OK in this drawing though based on how you updated the tests):

```
                +--------+                                  +------+
                |        |                                  |      |
                |        |                                  |      |
                |        |                                  |      |
                |        |                                  |      |
             17 |      18|                                25|    26|
            +---+--------v----+                       +-----+------v---+
            |                 |                       |                |
            |                 |                       |                |
15          |                 |11                   11|                | 22
------------+      sw1        +-----------------------+                +----------
            |                 |                       |     sw6        |
            |                 |                       |                |
            |                 |                       |                |
            |                 |                       |                |
            +-----------------+                       +----------------+
```

Reply:
Ok.